### ISSUE TITLE
Memory leak when multiInstance set to true

### ISSUE THREAD

**Issue by erguotou520** [pubkey/rxdb#778]
## Case
It got memery leak when `multiInstance` set to true.

## Issue
I just create a `RxDB` instance with `multiInstance` to be true in the Electron page.
And do nothing but just record the `Performance` in devtool.
Here is the memery usage:

![image](https://user-images.githubusercontent.com/7945757/44843739-91b03800-ac7b-11e8-9e55-35b1cf66a816.png)

It seems the the problem code is in `src/socket.js`'s `pull` loop function.

## Info
  - Environment: Electron@2.0.7
  - Adapter: pouchdb-adapter-idb@7.0.0
  - Stack: [https://github.com/SimulatedGREG/electron-vue](https://github.com/SimulatedGREG/electron-vue)

## Code

```js
  const db = await RxDB.create({
    name: 'test',
    adapter: 'idb',
    multiInstance: true
  })
  await db.collection({
    name: 'heroes',
    schema
  })
  console.log(db)
```

And the schema is: 

```js
export default {
  'title': 'hero schema',
  'version': 0,
  'description': 'describes a simple hero',
  'type': 'object',
  'properties': {
    'name': {
      'type': 'string',
      'primary': true
    },
    'color': {
      'type': 'string'
    },
    'healthpoints': {
      'type': 'number',
      'min': 0,
      'max': 100
    },
    'secret': {
      'type': 'string'
    },
    'birthyear': {
      'type': 'number',
      'final': true,
      'min': 1900,
      'max': 2050
    },
    'skills': {
      'type': 'array',
      'maxItems': 5,
      'uniqueItems': true,
      'item': {
        'type': 'object',
        'properties': {
          'name': {
            'type': 'string'
          },
          'damage': {
            'type': 'number'
          }
        }
      }
    }
  },
  'required': ['color']
}
```

**Comment by erguotou520** [comment]
The memery can't be released until click the `Collect garbage` button.

**Comment by pubkey** [comment]
@erguotou520 this type of memory-leak was reported before, but noone send a reproducible test.

The whole multiInstance-handling was rewritten for version 8.0.0 to not use the hacky socket-workarround [see here](https://github.com/pubkey/rxdb/blob/8.0.0/orga/releases/8.0.0.md).

I do not think it's worth the time to fix this, because the next major-release will come soon.
However a PR with a fix for this is welcomed if you need a quick solution.

**Comment by erguotou520** [comment]
When will `8.0.0` be stable release?

**Comment by pubkey** [comment]
I do not know. I planned to release the beta on the weekend. If there are no big issues it will take another week to release the stable version.

**Comment by erguotou520** [comment]
I think the workaround is expose `gc` function to the `global` and call `global.gc && global.gc()` in `pull` function?

**Comment by pubkey** [comment]
@erguotou520 have you checked if this problem appears also in the 8.0.0-beta?

**Comment by erguotou520** [comment]
Not successfully migrate to 8.0.0 😞

**Comment by pubkey** [comment]
I just released `8.0.0` which should fix this. Reopen if not.
